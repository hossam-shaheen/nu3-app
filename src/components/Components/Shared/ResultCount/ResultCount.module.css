.result-count {
  margin: 0;
}
