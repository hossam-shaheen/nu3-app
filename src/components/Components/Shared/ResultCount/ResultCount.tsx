import { FunctionComponent } from "react";
import classes from "./ResultCount.module.css";

export const ResultCount: FunctionComponent<{ count: number }> = ({
  count,
}): JSX.Element => {
  return (
    <p className={classes["result-count"]}>
      <span>
        {count} {count > 1 ? "results" : "result"}{" "}
      </span>
    </p>
  );
};
export default ResultCount;
