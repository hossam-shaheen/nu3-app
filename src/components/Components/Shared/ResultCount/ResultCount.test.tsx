import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import ResultCount from "./ResultCount";

test("ResultCount pass count 0 : Renders", () => {
  render(<ResultCount count={0} />);
  const resultText = screen.getByText("0 result");
  expect(resultText).toBeInTheDocument();
});

test("ResultCount pass count 1 : Renders", () => {
  render(<ResultCount count={1} />);
  const resultText = screen.getByText("1 result");
  expect(resultText).toBeInTheDocument();
});

test("ResultCount pass count 2 : Renders", () => {
  render(<ResultCount count={2} />);
  const resultText = screen.getByText("2 results");
  expect(resultText).toBeInTheDocument();
});

test("ResultCount pass count 0: Snapshot", () => {
  const component = renderer.create(<ResultCount count={0} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("ResultCount pass count 1: Snapshot", () => {
  const component = renderer.create(<ResultCount count={1} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("ResultCount pass count 2: Snapshot", () => {
  const component = renderer.create(<ResultCount count={2} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
