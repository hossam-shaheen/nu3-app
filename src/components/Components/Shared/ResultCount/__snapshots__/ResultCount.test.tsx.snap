// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCount pass count 0: Snapshot 1`] = `
<p
  className="result-count"
>
  <span>
    0
     
    result
      
  </span>
</p>
`;

exports[`ResultCount pass count 1: Snapshot 1`] = `
<p
  className="result-count"
>
  <span>
    1
     
    result
      
  </span>
</p>
`;

exports[`ResultCount pass count 2: Snapshot 1`] = `
<p
  className="result-count"
>
  <span>
    2
     
    results
      
  </span>
</p>
`;
